### PR TITLE
Reland "Remove all service extensions from release mode (#23038)"

### PR DESCRIPTION
**Pull request by goderbauer**
This reverts commit bb0290a4dc2ebef037bdc3c9d9e175c631b40f41.

### CODE REVIEW

**Comment by goderbauer** [comment]
To be landed after https://github.com/flutter/flutter/pull/23289 has collected some (stabilized) benchmark numbers.

File: packages/flutter/lib/src/widgets/binding.dart [hunk 1]
```diff
@@ -265,63 +265,55 @@ mixin WidgetsBinding on BindingBase, SchedulerBinding, GestureBinding, RendererB
...
         setter: (bool value) async {
           if (debugProfileBuildsEnabled != value)
             debugProfileBuildsEnabled = value;
-        }
+        },
       );
-      return true;
-    }());
 
-    // This service extension is deprecated and will be removed by 7/1/2018.
```
Comment (jacob314) [inline packages/flutter/lib/src/widgets/binding.dart]: go ahead and leave this service extension in for now and I'll revert my CL to add it back.
It is fine for the extension to be inside a assert(() ... block.

**Review by jacob314 — APPROVED** [review]

File: packages/flutter/lib/src/widgets/binding.dart (hunk 1 above)
Comment (goderbauer) [reply packages/flutter/lib/src/widgets/binding.dart]: Done.

**Comment by goderbauer** [comment]
All checks are actually green on Cirrus, they just haven't been reported back to github properly.

**Pull request merged**